### PR TITLE
FRONTEND-7011 :: Feature Internal :: Upgrade libraries/dependencies from Harmony TypeScript

### PR DESCRIPTION
**Pull request by orioljp**
**Asana task link:**
https://app.asana.com/0/0/1203311746789643/f

**Pull request information:**
Dependencies update

### CODE REVIEW

**Review by joselufo — APPROVED** [review]
Happy to approve it, but @orioljp are we sure we want to add * to all the minor versions?

**Review by Alex-DA — APPROVED** [review]

**Comment by Alex-DA** [comment]
> Happy to approve it, but @orioljp are we sure we want to add * to all the minor versions?

We are using this strategy in Backend also, IMO it's better that use the `^`.

File: packages/nest/package.json [hunk 1]
```diff
@@ -30,10 +30,10 @@
     "peerDependencies": {
         "@mobilejazz/harmony-core": "*",
         "@mobilejazz/harmony-typeorm": "*",
-        "@nestjs/common": "^9.0.0"
+        "@nestjs/common": "9.2.*"
```
Comment (doup) [inline packages/nest/package.json:33]: You've done [similar changes in the reference](https://github.com/mobilejazz/harmony-reference/pull/133/files); in this case though you *shouldn't* do the same. You're over-constraining Nest.

- `9.2.*` only allows patch updates (same as `~9.2.0`). So if our project uses anything that isn't `9.2.x` we'll get a warning/error. These won't be allowed: `9.0.x`, `9.1.x`, `9.3.x`…
  - So… we can't upgrade to `9.3.x` until Harmony changes its constraints.
- In this case the previous one was OK: `^9.0.0`.

File: packages/angular/package.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
         "prepublishOnly": "cp ./package.json ./dist && cp ../../README.md ./dist"
     },
     "peerDependencies": {
-        "@angular/common": ">=12",
+        "@angular/common": ">=13",
```
Comment (doup) [inline packages/angular/package.json:23]: Today while updating `eslint-config` I had to change this, as it wasn't working. [The change](https://github.com/mobilejazz/eslint-config/commit/a170c6187395af0b4772ed2cfe63c78488990bd5) follows the suggestion from here: [Make your NPM package support multiple versions of peer dependency](https://stackoverflow.com/questions/47309598/make-your-npm-package-support-multiple-versions-of-peer-dependency).

```suggestion
        "@angular/common": "13 - 15",
```

We could also do what some other libraries do: `^13.0.0 || ^14.0.0 || ^15.0.0`. But it's extra verbose and you need to remember what `^` does (allow different minor/patch).

I've created a task earlier, so now that you're with this I'm assigning it to you: https://app.asana.com/0/1109863238977521/1203409192207937/f

**Review by doup — CHANGES_REQUESTED** [review]
Please review the versions in general, I think we're doing this wrong. Today I had to change version constraints in `eslint-config` because we're using similar patterns.

We must assume that the community will respect semver, so we allow minor/patch versions of our required packages. That's why:

- If we allow a single major, use `^` (which allows higher minor/patch). If we know we need some version but not lower, then pin on that, e.g. `^2.3.1` our code works with `2.3.1`, but not lower. We assume it should work with higher minor/patch versions. Here I think we should use the lowest possible inside the major, eg: `^2.0.0` is better than `^2.3.1` if `2.0.0` still works.
- ~If we allow multiple majors (useful for Angular or Nest)~
  - ~Use simple range: `min - max` (both inclusive). E.g. `13 - 15`, this includes: 13, 14, 15.~
  - If we know that our code doesn't work with a version inside a major, but it works with later minor/patch, then use: `^X.Y.Z || ^U.V.W || ...`.
    - e.g. `^13.2.0 || ^14.0.0 || ^15.0.0`. Our code doesn't work if 13 is below `13.2.0`. It only from that version up. And in all versions of 14 and 15. If there isn't any minor/patch constraint, then we should favor the simplified range mode: `13 - 15`.

What do you think?

Related:
- https://stackoverflow.com/questions/22343224/whats-the-difference-between-tilde-and-caret-in-package-json
- https://stackoverflow.com/questions/47309598/make-your-npm-package-support-multiple-versions-of-peer-dependency

---

**Edit: by https://semver.npmjs.com `>=13` is perfectly fine.**

File: packages/angular/package.json (hunk 2 above)
Comment (doup) [reply packages/angular/package.json:23]: Actually, I'm not sure about this… I've just double checked in: https://semver.npmjs.com And `>=13` should be OK. I'm not sure what was the issue I was having in `esling-config`.

**Comment by lucianosantana** [comment]
I agree with @doup

File: packages/nest/package.json (hunk 1 above)
Comment (urijp) [reply packages/nest/package.json:33]: @doup you're right. Totally. I upgraded all dependencies as if it was a project but it shold allow projects with older versions to use it. I'll update this.

**Pull request closed without merging**